### DISCUSSION TITLE
Some tools that you may find useful

### DISCUSSION THREAD

**Discussion by taschenlampe** [openSUSE/openSUSEway#67] [Ideas]
Hi!

I know that configuring your system is kind of very user specific, still I´d like to share some adjustments I made, that may be useful for others:

*[Autotiling](https://pypi.org/project/autotiling/) 
I know this is not something all i3/sway users will like, as it is not the default way how sway works, nonetheless I find this tool to be very cool. 

*[Udiskie](https://github.com/coldfix/udiskie) 
This little tool allows to manage storage devices like usb-disk/cellphones....it populates your waybar with a little tray icon -> useful

*[Grim](https://github.com/emersion/grim) + [Slurp](https://github.com/emersion/slurp)
These tools I use to create screenshot tool. This is how I´ve set it up
```
# Screen capture 
set $ps1 Print
set $ps2 Control+Print
set $ps3 Alt+Print
set $ps4 Alt+Control+Print 
bindsym $ps1 exec grim
bindsym $ps2 exec grim -g "$(slurp)" 
bindsym $ps3 exec grim $(xdg-user-dir PICTURES)/screenshot_$(date +"%Y%m%d%H%M%S").png
bindsym $ps4 execgrim -g "$(slurp)" $(xdg-user-dir PICTURES)/ps_$(date +"%Y%m%d%H%M%S").png
```

*Resize
The way that sway resizes windows is somehow "clumsy", first you need to enable the "resize" mode and then only you may resize. Ok, this is ok, but there is a way to do it directly:
```
bindsym    $mod+Control+$left   resize shrink width 40px 
bindsym    $mod+Control+$down   resize shrink height 40px
bindsym    $mod+Control+$up     resize grow height 40px
bindsym    $mod+Control+$right  resize grow width 40px 
bindsym    $mod+Control+Left   resize shrink width 40px 
bindsym    $mod+Control+Down   resize shrink height 40px  
bindsym    $mod+Control+Up     resize grow height 40px 
bindsym    $mod+Control+Right  resize grow width 40px
```
*Tab
Use mod+Tab to toggle through your workspaces
```
#Toggle Tab 
bindsym $mod+tab workspace next_on_output  
bindsym $mod+Shift+tab workspace prev_on_output 
```

* Touchpad
This could be useful for laptop users
```
#Touchpad
 input tgype:touchpad { 
    tap enabled
    natural_scroll enabled
 }
```

As I wrote, all of this are of course very personal choices, still I hope that some of you may find this or that useful.
Let me know what you think of them, maybe some of these tool may be added to the openSuseway.

Cheers!

**Comment by FilippoBonazziSUSE** [comment]
About these points:

- Autotiling: I don't think we want this as default. But might be interesting as an option for user configuration
- Udiskie: I am not convinced we need this as default, but I would be open to it if there is interest. I personally don't use it.
- Screenshot configuration: a screenshot menu was added in #131.
- Resizing: I agree the menu-based resizing is a bit clumsy. However, it looks to me like this proposal has the same issues. Furthermore, I only ever resize windows with the mouse, so I don't know how useful it would be to add this as a second default other than the resizing menu.
- Workspace switching: this was done in #73, and I also recently added support for switching workspaces via 3-finger swipes on a touchpad in #211.
- Touchpad: this was also done in #73.
